smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/integration/HttpRpcProtocolGenerator.java

```diff
@@ -193,9 +193,12 @@ private void writeRequestHeaders(GenerationContext context, OperationShape opera
 
         // The Content-Type header is always present.
         writer.addImport("HeaderBag", "__HeaderBag", "@aws-sdk/types");
-        writer.write("const headers: __HeaderBag = {};");
-        writer.write("headers['Content-Type'] = $S;", getDocumentContentType());
-        writeDefaultHeaders(context, operation);
+        writer.openBlock("const headers: __HeaderBag = {", "};",
+            () -> {
+                writer.write("'Content-Type': $S,", getDocumentContentType());
+                writeDefaultHeaders(context, operation);
```

Comment:
See note on other file about the contract change to `writeDefaultHeaders`, as it applies here too.

Reply:
Documentation updated in https://github.com/awslabs/smithy-typescript/pull/175/commits/69bd79e3e6b51f4302ae63e5d6a177160328609b

The downstream implementation was updated in CodeGen commit https://github.com/aws/aws-sdk-js-v3/pull/1134/commits/a121ebea1cd00442e11e95cfeccf81f11bca6403 of linked PR https://github.com/aws/aws-sdk-js-v3/pull/1134